enhancements/olm/platform-operators.md

```diff
@@ -0,0 +1,609 @@
+participant ClusterOperator
+participant PO manager
+Installer->>Admin: Creates the Kubernetes manifests
+CVO->>PO custom resource: Applies
+Admin-->>Installer: Edits Kubernetes manifests
+Installer->>Bootstrap: Creates
+Bootstrap->>CVO: Deploys
+CVO->>PO manager: Deploys
+PO manager->>ClusterOperator: Manages
+PO manager->>PO custom resource: Watches
+CVO->>ClusterOperator: Watches
+```
+
+Note: The above workflow requires admins to manually create, and edit, the cluster Kubernetes manifests with their desired PO intent. There's a chance that admins mistype the package names, and depending on whether upfront validation is possible, that invalid configuration may not be surfaced until after cluster creation. This behavior may require explicit documentation to reduce this risk.
+These are obvious UX gaps with this workflow, which can be solidified in future phases, but it will result in zero installer changes during the 4.12 timeframe.
+
+### API Extensions
+
+When the "TechPreviewNoUpgrades" feature set is enabled for a cluster, several API extensions will be added during phase 0.
+
+#### OpenShift API(s)
+
+> Note: The following API(s) will live in the openshift/api repository.
+
+- Introduces a top-level PlatformOperator custom-scoped v1alpha1 CRD:
+
+```go
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+//+kubebuilder:resource:scope=Cluster
+
+// PlatformOperator is the Schema for the platformoperators API.
+type PlatformOperator struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    Spec   PlatformOperatorSpec   `json:"spec,omitempty"`
+    Status PlatformOperatorStatus `json:"status,omitempty"`
+}
+
+// PlatformOperatorSpec defines the desired state of PlatformOperator.
+type PlatformOperatorSpec struct {
+    // PackageName contains the desired OLM-based Operator package name
+    // that is defined in an existing CatalogSource resource in the cluster.
+    //
+    // This configured package will be managed with the cluster's lifecycle.
+    // +kubebuilder:validation:Required
+    PackageName string `json:"packageName"`
+}
+
+// PlatformOperatorStatus defines the observed state of PlatformOperator.
+type PlatformOperatorStatus struct {
+    // Conditions represent the latest available observations of a platform operator's current state.
+    // +optional
+    Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+```
+
+#### Upstream OLM 1.x API(s)
+
+> Note: The following API(s) will live in the openshift/operator-framework-rukpak repository.
+
+- [Introduces the upstream Bundle and BundleDeployment OLM 1.x v1alpha1 API(s)](https://github.com/operator-framework/rukpak/tree/main/api/v1alpha1).
+- [Introduces a validating webhook for the Bundle API](https://github.com/operator-framework/rukpak/blob/main/api/v1alpha1/bundle_webhook.go).
+- [Introduces a validation webhook for CRDs](https://github.com/operator-framework/rukpak/tree/main/cmd/crdvalidator).
```

Comment:
Should we leave the CRD validator out? I know we've been debating that even from an upstream rukpak perspective.

Reply:
Updated in https://github.com/openshift/enhancements/pull/1170/commits/bc14b4537125d43153cc452cd686e02cf96630b2.